### PR TITLE
[MLIR][docs] Mention declarePromisedInterface in Interfaces doc

### PR DESCRIPTION
**Pull request by frederik-h**

### CODE REVIEW

**Review by ftynse — APPROVED** [review]

**Comment by ftynse** [comment]
Would be helpful to list what the message looks like so quick googling references to this page.

**Comment by frederik-h** [comment]
> Would be helpful to list what the message looks like so quick googling references to this page.

Thanks for the review. I have now added an example of the error message.

**Review by joker-eph — APPROVED** [review]
Thanks!

File: mlir/docs/Interfaces.md [hunk 1]
```diff
@@ -299,6 +299,23 @@ owner of the dialect containing the object nor the owner of the interface are
...
+    declarePromisedInterface<SomeInterface, SomeOp>();
+     ...
+  }
+```
+Now attempting to use the interface, e.g in a cast, without a prior
+registration of the external model will lead to a runtime error that will
+look similar to this:
+```
+LLVM ERROR: checking for an interface (`SomeInterface`) that was promised by dialect 'mydialect' but never implemented. This is generally an indication that the dialect extension implementing the interface was never registered.
+```
```
Comment (joker-eph) [inline mlir/docs/Interfaces.md:317]: Maybe add:

```suggestion
```

If you encounter this error for a dialect and an interface provided by MLIR, you may look for a method that will be named like `register<Dialect><Interface>ExternalModels(DialectRegistry &registry);` ; try to find it with `git grep 'register.*SomeInterface.*Model' mlir`. 
```

**Review by joker-eph — APPROVED** [review]
Can you please rewrap the doc to 80 columns though?

In general I use `mdformat --wrap 80` for this.

File: mlir/docs/Interfaces.md (hunk 1 above)
Comment (frederik-h) [reply mlir/docs/Interfaces.md:319]: > Maybe add:

That should be useful! Thanks for the addition.

**Comment by frederik-h** [comment]
> Can you please rewrap the doc to 80 columns though?
> 
> In general I use `mdformat --wrap 80` for this.

I have only rewrapped the new changes introduced by this PR since applying it to the whole document would introduce lots of unrelated changes, i.e. the rest of the document is not formatted in that way yet.

**Pull request merged**